amazon/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/amazon/AmazonAuthenticator.java

```diff
@@ -0,0 +1,226 @@
+        callbackUrl.setDisplayOrder(2);
+        configProperties.add(callbackUrl);
+        return configProperties;
+    }
+
+
+    /**
+     * Get OAuth2 Scope
+     *
+     * @param scope                   Scope
+     * @param authenticatorProperties Authentication properties.
+     * @return OAuth2 Scope
+     */
+    @Override
+    protected String getScope(String scope, Map<String, String> authenticatorProperties) {
+        return AmazonAuthenticatorConstants.AMAZON_SCOPE_PROFILE;
+    }
+
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+            String clientSecret = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET);
+            String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+            String callbackUrl = getCallbackUrl(authenticatorProperties);
+            OAuthAuthzResponse authorizationResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            String code = authorizationResponse.getCode();
+            OAuthClientRequest accessRequest =
+                    getAccessRequest(tokenEndPoint, clientId, code, clientSecret, callbackUrl);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getOauthResponse(oAuthClient, accessRequest);
+            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            if (StringUtils.isBlank(accessToken)) {
+                throw new AuthenticationFailedException("Access token is empty or null");
+            }
+            context.setProperty(OIDCAuthenticatorConstants.ACCESS_TOKEN, accessToken);
+            Map<ClaimMapping, String> claims;
+            AuthenticatedUser authenticatedUserObj;
+            String json = sendRequest(AmazonAuthenticatorConstants.AMAZON_USERINFO_ENDPOINT,
+                    oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN));
+            JSONObject obj = new JSONObject(json);
+            authenticatedUserObj = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier((String) obj.get(AmazonAuthenticatorConstants.USER_ID));
+            authenticatedUserObj.setAuthenticatedSubjectIdentifier((String) obj.get(AmazonAuthenticatorConstants.USER_ID));
+            claims = getSubjectAttributes(oAuthResponse, authenticatorProperties);
+            authenticatedUserObj.setUserAttributes(claims);
+            context.setSubject(authenticatedUserObj);
+        } catch (OAuthProblemException | IOException | JSONException e) {
```

Comment:
We need to have Java 1.7 or above to use this feature. Please add README and specify the pre-requisites.